src/lib/y2security/autoinst/lsm_config_reader.rb

```diff
@@ -0,0 +1,58 @@
+require "y2security/lsm/config"
+
+module Y2Security
+  module Autoinst
+    # This class is responsible of reading the LSM configuration declared in the AutoYaST profile
+    class LSMConfigReader
+      attr_reader :section
+
+      # Constructor
+      #
+      # @param section [AutoinstProfile::LSM::LsmSection]
+      def initialize(section)
+        @section = section
+      end
+
+      # Reads the LSM configuration defined in the profile modifying it accordingly
+      def read
+        return unless section
+
+        config.configurable = section.configurable
+        config.select(section.select) if section.select
+
+        configure_selinux(section.selinux) if section.selinux
```

Comment:
NP: maybe you can use `configure_xyz` only and then early return if there are no SELinux section. I mean

```ruby
def read
  ...
  configure_selinux
  configure_apparmor
  ...
end

...

private

attr_reader :section

def configure_selinux
  return unless section.selinux
  ...
end
```

Reply:
Yep, in fact I was thinking even in some configure_supported wich iterates over each of the modules checking if the section is present and the config exist for that module ... but by now I leave it more explicit